bazel/linting/scripts/lint_go.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
```

Comment:
a top level comment? what does this script do?

also, which global environment variables does it expect? not immediately obvious. Looks like $TARGET? $GO_LIBRARY_NAME?

I would consider also adding ifs/conditionals erroring out if those variables are not provided.